docs/prover/cli/options.md

```diff
@@ -1231,6 +1231,34 @@ The second use is when the solvers can prove the property, they just need more t
 certoraRun Bank.sol --verify Bank:Bank.spec --smt_timeout 300
 ```
 
+(--max_concurrent_rules)=
+## `max_concurrent_rules`
+
+**Usage**
+```sh
+--max_concurrent_rules <n>
+```
+
+**What does it do?**
+
+This attribute controls the maximum number of rule evaluations that can be executed concurrently.
+By default, it is set to the number of available CPU cores on the host machine, allowing optimal parallelization under typical conditions.
+Setting this value limits the number of rule evaluation tasks that can run in parallel.
+By capping the concurrency level, this attribute helps regulate system resource usage, particularly memory consumption, 
+and can prevent resource exhaustion.
+
+
+**When to use it?**
+
+Use this attribute when you are encountering out-of-memory errors, particularly when processing unusually large or complex rule sets.
+Lowering the concurrency can reduce peak memory usage by staggering evaluation workloads.
```

Comment:
What is the default concurrency level on the cloud on production?
Is it a good idea to write it down in our documentation? @H00N24

Reply:
I don't see any cloud-based modifications for this flag, so it should be the prover's default value.

Reply:
By default, we have 18 vCPUs available.